speechbrain/decoders/seq2seq.py

```diff
@@ -1313,6 +1386,111 @@ def lm_forward_step(self, inp_tokens, memory):
         return log_probs[:, -1, :], memory
 
 
+class S2SWhisperBeamSearch(S2SBeamSearcher):
+    """This class implements the beam search decoding
+    for Whisper neural nets made by OpenAI in
+    https://cdn.openai.com/papers/whisper.pdf.
+
+    Arguments
+    ---------
+    module : list with the followings one:
+        model : torch.nn.Module
+            A whisper model. It should have a decode() method.
+        ctc_lin : torch.nn.Module (optional)
+            A linear output layer for CTC.
+    **kwargs
+        Arguments to pass to S2SBeamSearcher
+    """
+
+    def __init__(
+        self, module, temperature=1.0, temperature_lm=1.0, **kwargs,
+    ):
+        super(S2SWhisperBeamSearch, self).__init__(**kwargs)
+
+        self.model = module[0]
+        if len(module) == 2:
+            self.ctc_fc = module[1]
+
+        self.softmax = torch.nn.LogSoftmax(dim=-1)
+
+        self.temperature = temperature
+        self.temperature_lm = temperature_lm
+
+        self.decoder_input_tokens = None
+        self.language_token = 50259  # default language is english
```

Comment:
Can't we have this as arguments instead? Hard coding it in the init is WOW.

Reply:
We can. It was just a default value instead of settings None. The value should be overridden by the setters. I don't fully like the idea of setting it directly in the yaml file as the value will be anyway modified with the setters. We could instead give a default value in args (e.g language_token=None or language_token=50259) instead of hardcoding it in the init. (but it's just a matter of where are we hardcoding the value :p)

Reply:
> Once the requested changes will be done and results added, the PR will be ready to be merger.

Are there public pre-trained models for this PR?

Reply:
@csukuangfj we are training models, and we will certainly release them. It is most likely that we will first merge then release (to give code for the HuggingFace fine-tuning challenge). I don't know either if we will provide an inference interface, but maybe ...